homeassistant/components/overkiz/config_flow.py

```diff
@@ -1,78 +1,221 @@
+            )
+
+            # Verify SSL blocked by https://github.com/Somfy-Developer/Somfy-TaHoma-Developer-Mode/issues/5
+            # Somfy (self-signed) SSL cert uses the wrong common name
+            session = async_create_clientsession(self.hass, verify_ssl=False)
+
+            # Local API
+            local_client = OverkizClient(
+                username="",
+                password="",
+                token=token,
+                session=session,
+                server=generate_local_server(host=user_input[CONF_HOST]),
+            )
+
+            try:
+                await local_client.login()
+            except Exception as exception:  # pylint: disable=broad-except
+                try:
+                    # Remove local token when login is not successful
+                    await client.delete_local_token(gateway_id, uuid)
+                except NotSuchTokenException:
+                    LOGGER.debug("Token is invalid / has been already removed")
+
+                raise exception
 
-        await client.login(register_event_listener=False)
+            user_input[CONF_TOKEN] = token
+            user_input[CONF_TOKEN_UUID] = uuid
 
-        # Set first gateway id as unique id
+        else:
+            server = SUPPORTED_SERVERS[user_input[CONF_SERVER]]
+
+            session = async_create_clientsession(self.hass)
+            client = OverkizClient(
+                username=username, password=password, server=server, session=session
+            )
+
+            await client.login(register_event_listener=False)
+
+        # Set main gateway id as unique id
         if gateways := await client.get_gateways():
-            gateway_id = gateways[0].id
-            await self.async_set_unique_id(gateway_id)
+            for gateway in gateways:
+                if is_overkiz_gateway(gateway.id):
+                    gateway_id = gateway.id
+                    await self.async_set_unique_id(gateway_id)
+
+        return user_input
 
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle the initial step via config flow."""
-        errors = {}
+        errors: dict[str, str] = {}
+
+        if user_input:
+            self._server = user_input[CONF_SERVER]
+
+            # Some Overkiz hubs do support a local API
+            # Users can choose between local or cloud API.
+            if self._server in SERVERS_WITH_LOCAL_API:
+                return await self.async_step_local_or_cloud()
+
+            return await self.async_step_cloud()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_SERVER, default=self._server): vol.In(
+                        {key: hub.name for key, hub in SUPPORTED_SERVERS.items()}
+                    ),
+                }
+            ),
```

Comment:
```suggestion
        servers = {key: hub.name for key, hub in SUPPORTED_SERVERS.items()}
        return self.async_show_form(
            step_id="user",
            data_schema=self.add_suggested_values_to_schema(
                vol.Schema(
                    {
                        vol.Required(CONF_SERVER): vol.In(servers),
                    }
                ),
                {CONF_SERVER: self._server},
            ),
```

Reply:
What would be the benefit of using `add_suggested_values_to_schema()`? And of moving the `default=` to another notation?

Reply:
@joostlek?